lib/cache/src/filesystem.rs

```diff
@@ -98,7 +98,13 @@ impl Cache for FileSystemCache {
             key.to_string()
         };
         let path = self.path.join(filename);
-        Module::deserialize_from_file(store, path)
+        let ret = Module::deserialize_from_file(store, path.clone());
+        if ret.is_err() {
+            // If an error occurs while deserializing then we can not trust it anymore
+            // so delete the cache file
+            let _ = std::fs::remove_file(path);
+        }
+        ret
```

Comment:
I'm not sure we should handle this here.
Basically, let's say I put a file by mistake (that is not wasm) and try to load it. This will remove it.
We don't want that

Reply:
It makes sense that the cache path is owned , we add and delete things in there anyway.

So I'm resolving.